### PR TITLE
FIX: anatomical discrete segmentation improvements

### PR DESCRIPTION
**Pull request by eilidhmacnicol**
## Changes proposed in this pull request
Reducing the shrink factors of the bias field correction improves anatomical segmentation.
Also small refactor of FAST to make sure bias field correction is not run by FAST as well as N4.
FAST outputs are also adjusted.

Depends on #40

### CODE REVIEW

File: fprodents/interfaces/patches.py [hunk 1]
```diff
@@ -58,3 +60,30 @@ def _run_interface(self, runtime):
...
 
         return runtime
+
+
+class _FixTraitFASTInputSpec(FASTInputSpec):
+    bias_iters = traits.Range(
+        low=0,
+        high=10,
+        argstr="-I %d",
+        desc="number of main-loop iterations during " "bias-field removal",
```
Comment (oesteban) [inline fprodents/interfaces/patches.py:70]:
```suggestion
        desc="number of main-loop iterations during bias-field removal",
```

File: fprodents/patch/workflows/anatomical.py [hunk 2]
```diff
@@ -423,16 +423,16 @@ def _check_img(img):
...
+        ),
         name="anat_dseg",
         mem_gb=3,
     )
-    # Change LookUp Table - BIDS wants: 0 (bg), 1 (gm), 2 (wm), 3 (csf)
-    lut_anat_dseg = pe.Node(
-        niu.Function(function=_apply_bids_lut), name="lut_anat_dseg"
-    )
 
-    lut_anat_dseg.inputs.lut = (0, 3, 2, 1)  # Maps: 0 -> 0, 3 -> 1, 2 -> 2, 1 -> 3.
```
Comment (oesteban) [inline fprodents/patch/workflows/anatomical.py:435]: If there output labels are already what we expect (comment in line 430), then let's make this node do nothing
```suggestion
    lut_anat_dseg.inputs.lut = (0, 1, 2, 3)  # Maps: 0 -> 0, 3 -> 1, 2 -> 2, 1 -> 3.
```

File: fprodents/patch/workflows/anatomical.py [hunk 3]
```diff
@@ -423,16 +423,16 @@ def _check_img(img):
...
+            bias_iters=0,
+            no_bias=True,
+        ),
         name="anat_dseg",
         mem_gb=3,
     )
-    # Change LookUp Table - BIDS wants: 0 (bg), 1 (gm), 2 (wm), 3 (csf)
-    lut_anat_dseg = pe.Node(
-        niu.Function(function=_apply_bids_lut), name="lut_anat_dseg"
-    )
```
Comment (oesteban) [inline fprodents/patch/workflows/anatomical.py:433]:
```suggestion
    # Change LookUp Table - BIDS wants: 0 (bg), 1 (gm), 2 (wm), 3 (csf)
    lut_anat_dseg = pe.Node(
        niu.Function(function=_apply_bids_lut), name="lut_anat_dseg"
    )
```

File: fprodents/patch/workflows/anatomical.py [hunk 4]
```diff
@@ -472,8 +472,7 @@ def _check_img(img):
         (xfm_wm, mrg_tpms, [('output_image', 'in2')]),
         (xfm_csf, mrg_tpms, [('output_image', 'in3')]),
         (mrg_tpms, anat_dseg, [('out', 'other_priors')]),
-        (anat_dseg, lut_anat_dseg, [('partial_volume_map', 'in_dseg')]),
-        (lut_anat_dseg, outputnode, [('out', 't2w_dseg')]),
+        (anat_dseg, outputnode, [('partial_volume_map', 't2w_dseg')]),
```
Comment (oesteban) [inline fprodents/patch/workflows/anatomical.py:475]:
```suggestion
        (anat_dseg, lut_anat_dseg, [('partial_volume_map', 'in_dseg')]),
        (lut_anat_dseg, outputnode, [('out', 't2w_dseg')]),
```

File: fprodents/patch/workflows/anatomical.py [hunk 5]
```diff
@@ -482,7 +481,7 @@ def _check_img(img):
         ]),
         # step 5
         (anat_norm_wf, xfm_dseg, [('poutputnode.standardized', 'reference_image')]),
-        (lut_anat_dseg, xfm_dseg, [('out', 'input_image')]),
+        (anat_dseg, xfm_dseg, [('partial_volume_map', 'input_image')]),
```
Comment (oesteban) [inline fprodents/patch/workflows/anatomical.py:484]:
```suggestion
        (lut_anat_dseg, xfm_dseg, [('out', 'input_image')]),
```

File: fprodents/workflows/base.py [hunk 6]
```diff
@@ -306,15 +305,15 @@ def init_single_subject_wf(subject_id):
 
         bold_ref_wf = init_epi_reference_wf(
             auto_bold_nss=True,
-            omp_nthreads=config.nipype.omp_nthreads
+            omp_nthreads=config.nipype.omp_nthreads,
+            n4_iter=4,
+            adaptive_bspline_grid=True,
+            shrink_factor=1,
```
Comment (oesteban) [inline fprodents/workflows/base.py:311]: Please add some documentation about this decision on the docstring or as a comment above (or both)

File: fprodents/workflows/bold/base.py [hunk 7]
```diff
@@ -242,6 +242,7 @@ def init_func_preproc_wf(bold_file):
     outputnode = pe.Node(
         niu.IdentityInterface(
             fields=[
+                "bold_mask",
```
Comment (oesteban) [inline fprodents/workflows/bold/base.py:245]: I am guessing this comes from the other PR.

**Review by oesteban — APPROVED** [review]
LGTM, just a little nit pick attempting not to deviate too much from fmriprep

File: fprodents/patch/workflows/anatomical.py (hunk 2 above)
Comment (eilidhmacnicol) [reply fprodents/patch/workflows/anatomical.py:435]: I think this was actually based on some changes that I was testing (using FAST with T2 as the image type),so I've reverted it to it's original incarnation here

File: fprodents/workflows/base.py (hunk 6 above)
Comment (eilidhmacnicol) [reply fprodents/workflows/base.py:316]: Added a comment for now

**Pull request merged**